### PR TITLE
Week 6 zack

### PR DESCRIPTION
**Pull request by ZackRohde**
# Submitting Week 6 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [✅  ] Coding Challenges
- [✅  ] Coding Concepts
- [ ✅ ] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here: I felt great on the interview question and the rails commenting for the most part if you could correct anything that isnt correct that'd be great to study up on. For the JavaScript, the first one is completely over my head and knowledge I feel without copying code. I again think I understand the logic of majority of the question, but lack of experience is killing me.

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -23,24 +34,79 @@ const people = [
 
 // b) Create the function that makes the test pass.
 
+// PSUEDO CODE:
+// Create a function that takes in an array
+// Figure out how to convert an object into an array
+// I know I'm missing a lot here, I'm just not sure, this is a hard question....
+// I know well use string interpolation to access the name and occupation and use it for the return
```
Comment (elysemontano) [inline code-challenges.test.js]: Consider that you have an array of objects.  If you iterate on the array, you will be accessing an individual object which then you can utilize dot notation to access the values in the object.  If you get really stuck on a question, please reach out!  Don't suffer in silence.

File: code-challenges.test.js [hunk 2]
```diff
@@ -23,24 +34,79 @@ const people = [
...
 // b) Create the function that makes the test pass.
+
+// PSUEDO CODE:
+// Create a function and pass in an array
+// Cube each number 
+// I believe using .reduce() which takes in an array of numbers and returns a single value, will be used for this
+// Then we will take the value and the next value and add them together, remeber the reduce method will only output a single value. 
+// Console log both function names
+
+const sumOfNumCubed = (array) => {
```
Comment (elysemontano) [inline code-challenges.test.js:112]: Nice work on these last 2 problems!

File: interview-questions.md [hunk 3]
```diff
@@ -8,36 +8,70 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. As a developer, I am creating a Rails application with a model called Cohort that has_many students, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key? Would the foreign key be on the Cohort model or the Student model?
 
-Your answer:
+Your answer: To fix this you add the 1st model name_id:integer as a column. The name of the foreign key is 1st model_id and that is how they connect the two together. The foreign key would be on the student model as it's apart of that model which connects to the primiary key.
 
-Researched answer:
+Researched answer: I believe my way for fixing it is by adding a column, otherwise I'm unsure what to search to fix this mistake. Foreign keys and primary keys must always be connected so, the more tables you have they will all interconnect together.
```
Comment (elysemontano) [inline interview-questions.md:13]: You would want to generate a migration to be able to add a column.

File: interview-questions.md [hunk 4]
```diff
@@ -8,36 +8,70 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
+// Controller Method: Update and the action is you're updating exsisting data to the 2nd index
+
 action: "DELETE" location: /students/2
 
+// Controller Method: Destroy and your action is deleting index 2 for forever from the database
+
 5. As a developer, you are making an application to manage your to do list. Create 10 user stories that will help you get your application started. Read more about [user stories](https://www.atlassian.com/agile/project-management/user-stories).
+
+User Story 1: As a developer I want to create my application using my preffered languages.
```
Comment (elysemontano) [inline interview-questions.md:59]: Really excellent work in this section!

File: rails-commenting.rb [hunk 5]
```diff
@@ -7,24 +7,29 @@
...
   end
 
   # ---3)
+  # Were defining show which is a controller action. post is the instance variable. BlogPost is the class, and were finding a specific blog by the id.
   def show
     @post = BlogPost.find(params[:id])
   end
 
   # ---4)
+  # Were defining new which is a controller action. post is the instance variable. BlogPost is the class and the .new is giving the instance variable the ability to be viewed?
```
Comment (elysemontano) [inline rails-commenting.rb:25]: I am assuming the question mark here is asking for validation, and yes your are correct!

File: rails-commenting.rb [hunk 6]
```diff
@@ -7,24 +7,29 @@
 # FILE: app/controller/blog_posts_controller.rb
 
 # ---1)
+# Application Controller is the Parent Class and the BlogPostController is a Child Class.
```
Comment (elysemontano) [inline rails-commenting.rb:10]: Crushed this section!!

**Review by elysemontano — CHANGES_REQUESTED** [review]
Request changes: I would really like to see a little more on problem 1 in code challenges.  I left some feedback to consider in working through it.  If you are still struggling, please reach out and we can set up a time to work through this problem together.  

Other comments: All other sections were excellent!  I can certainly tell that you really shine with working on backend development concepts.  Keep that momentum up!

File: code-challenges.test.js (hunk 1 above)
Comment (ZackRohde) [reply code-challenges.test.js]: I believe I did it, but I can't get it to pass. It really helped understanding what was actually needed. I have a hard time understanding exactly what's being asked I find it too vague, I need more direction.

File: code-challenges.test.js [hunk 7]
```diff
@@ -23,24 +34,83 @@ const people = [
...
+// Create a function that takes in an array
+// Iterate over the different objects by using .map((object) because were iterating over the objects
+// Use string interpolation for the name in the object and occupation
+// Using .split("") we are individually seperating the elements
+// We must map over again to slice the (0,1) which is the first elements index and the second elements first index (I believe)
+// Slice the first letter of each word for the names by using object.name because we can use dot notation like you said to access the key value pairs
+// Capitalize it using .toUpperCase() method and join them back
+
+const capName = (array) => {
+  return array.map((object) => `${object.name.split("").map((firstName) => firstName.slice(0, 1).toUpperCase() + firstName.slice(1)).join("")} is ${object.occupation}.`
```
Comment (elysemontano) [inline code-challenges.test.js:47]: This looks great! Thank you for adding those changes!

**Review by elysemontano — APPROVED** [review]
Great work on the changes on problem 1!!

**Pull request merged**